packages/rum-core/src/domain/vital/vitalCollection.ts

```diff
@@ -11,9 +15,49 @@ export interface DurationVitalStop {
   stopClocks: ClocksState
 }
 
-export function startVitalCollection(lifecycle: LifeCycle) {
+interface Vital {
+  name: string
+  type: VitalType
+  startClocks: ClocksState
+  value: number
+}
+
+export function startVitalCollection(lifeCycle: LifeCycle) {
+  const vitalStartsByName: { [name: string]: DurationVitalStart } = {}
```

Comment:
**💬 suggestion:** when storing arbitrary keys in a hashmap, always prefer a `Map` over a plain object to avoid prototype polution and other issues when keys match native object properties.